test/helpers/logutils/utils.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2019 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logutils
+
+import (
+	"fmt"
+	"regexp"
+	"sort"
+	"strings"
+)
+
+const (
+	selfishThresholdMsg = "Goroutine took lock for more than" // from https://github.com/cilium/cilium/pull/5268
+
+	contextDeadlineExceeded = "context deadline exceeded"
+	errorLogs               = "level=error"
+	warningLogs             = "level=warning"
+	aPIPanicked             = "Cilium API handler panicked"
+)
+
+var countLogsMessages = []string{contextDeadlineExceeded, errorLogs, warningLogs, aPIPanicked, selfishThresholdMsg}
+
+// LogErrorsSummary returns error and warning summary for given logs
+func LogErrorsSummary(logs string) string {
+	var sb strings.Builder
+	for _, message := range countLogsMessages {
+		var prefix = ""
+		result := strings.Count(logs, message)
+		if result > 5 {
+			// Added a warning emoji just in case that are more than 5 warning in the logs.
+			prefix = "⚠️  "
+		}
+		fmt.Fprintf(&sb, "%sNumber of %q in logs: %d\n", prefix, message, result)
+	}
+	sb.WriteString("Top 5 errors/warnings:\n")
+
+	warns := getErrorWarningMsgs(logs, 5)
+	sb.WriteString(strings.Join(warns, "\n"))
+	return sb.String()
+}
+
+func getErrorWarningMsgs(logs string, n int) []string {
```

Comment:
Please add a short comment indicating what this function does for future code maintainers.